service/worker/scanner/tasklist/scavenger.go

```diff
@@ -257,11 +260,14 @@ func (s *Scavenger) process(taskListInfo *p.TaskListInfo) executor.TaskStatus {
 
 func (s *Scavenger) awaitExecutor() {
 	outstanding := s.executor.TaskCount()
+	timer := s.timeSource.NewTimer(s.pollInterval)
+	defer timer.Stop()
 	for outstanding > 0 {
 		select {
-		case <-time.After(s.pollInterval):
+		case <-timer.Chan():
 			outstanding = s.executor.TaskCount()
 			s.scope.UpdateGauge(metrics.TaskListOutstandingCount, float64(outstanding))
+			timer.Reset(s.pollInterval)
```

Comment:
probably another "likely better as a ticker" case, worth considering.  I doubt it's intentional that the delay happens only after TaskCount returns, that's just an atomic lookup.